src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -46,20 +46,24 @@
  (fn [{:keys [db]}]
    {:db (update-in db [:wallet :ui :send] dissoc :recipient :to-address)}))
 
-(rf/reg-event-fx :wallet/select-send-address
+(rf/reg-event-fx
+ :wallet/select-send-address
  (fn [{:keys [db]} [{:keys [address token recipient stack-id]}]]
    {:db (-> db
             (assoc-in [:wallet :ui :send :recipient] (or recipient address))
             (assoc-in [:wallet :ui :send :to-address] address))
     :fx [[:navigate-to-within-stack
           (if token [:wallet-send-input-amount stack-id] [:wallet-select-asset stack-id])]]}))
 
-(rf/reg-event-fx :wallet/send-select-token
+(rf/reg-event-fx
+ :wallet/send-select-token
  (fn [{:keys [db]} [{:keys [token stack-id]}]]
    {:db (-> db
             (update-in [:wallet :ui :send] dissoc :collectible)
             (assoc-in [:wallet :ui :send :token] token))
-    :fx [[:navigate-to-within-stack [:wallet-send-input-amount stack-id]]]}))
+    :fx [[:dispatch-later
+          {:ms       1
+           :dispatch [:navigate-to-within-stack [:wallet-send-input-amount stack-id]]}]]}))
```

Comment:
For some reason, when we navigate and read the `:token` value from a sub, re-frame doesn't have the value ready, it's `nil`, for following queries to the sub it is already populated.

Using `:dispatch-later` was the only one solution I found that worked, I also tried `:dispatch` as:
```clojure
:fx [[:dispatch [:navigate-to-within-stack [...]]]]
```
But it wasn't enough.

cc: @ilmotta 
Re-frame is supposed to first perform the `:db` update and then the `:fx` but it seems this is not the case :thinking: 

If this is happening here, I think this also happens in other screens, we noticed it because we are using this value to initialize a component (in a reagent form-2) in our case, a component was throws an exception if we didn't pass the expected value

Reply:
Yeah! I'll open the issue, I was thinking that maybe the subs were too many, and they take longer to propagate than the rendering of the screen (weird, since maybe the screen is heavier) :thinking:

Reply:
perhaps for now this deserves a comment to explain why :) (in the codebase)

Reply:
Yeah! I will add it @J-Son89

Reply:
@J-Son89  done!

Reply:
@ilmotta @J-Son89 

Just a quick update on this.

As reported by QA, seems this hack isn't working all the time, setting a custom timeout (like 16 or 32ms) is something I don't want to do, so I'll investigate the root cause or fix this issue in other ways than just delaying a dispatch.